Library/Homebrew/extend/os/linux/missing_formula.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Homebrew
+  module MissingFormula
+    class << self
+      def cask_reason(*)
+        # Casks aren't supported on non-MacOS.
```

Comment:
Sorry, I should have explained this better but I was hoping for the reverse e.g. in `os/mac/missing_formula.rb`. Thanks!

Reply:
Ah! 😳To be sure I understand the intended outcome for "reversing" this, I have a couple of questions as there are several approaches going around in my head:

- The [current `missing_formula.rb` `cask_reason` code](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/missing_formula.rb#L199-L208) is moved into the `extend/os/mac/missing_formula.rb` so that it's only called on MacOS, and for the [original method call](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/missing_formula.rb#L12) to be scoped to `OS.mac?`?
- The `extend/os/mac/missing_formula.rb` is changed to `return if OS.linux?`? But [the Mac extend file is only `require`d if already on MacOS](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/missing_formula.rb#L3), so Linux would never get there.
- Or, something else?

Reply:
> * The [current `missing_formula.rb` `cask_reason` code](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/missing_formula.rb#L199-L208) is moved into the `extend/os/mac/missing_formula.rb` so that it's only called on MacOS

Yup, this, thanks!

Reply:
Thanks for confirming. I've tested manually that it still works as intended on both MacOS and Linux, and pushed the new changes.